src/controllers/FacetQueryController.ts

```diff
@@ -327,15 +329,6 @@ export class FacetQueryController {
     return mergeWith;
   }
 
-  private processQueryOverrideForBasicExpression(queryBuilder: QueryBuilder, mergeWith: QueryBuilderExpression) {
-    // If any part of the query override built so far inside `mergeWith` is not empty
-    // we must ensure that the basic expression is copied properly
-    if (Utils.isEmptyString(mergeWith.basic) && mergeWith.full != '') {
```

Comment:
Is the second part of the condition needed? Not sure what it is for.
```
mergeWith.full != ''
```

Reply:
The `full` is all expressions overrides together I believe. That's why he checked there that  
> If any part of the query override built so far inside `mergeWith` is not empty
but in the case of the additional filter we always want to pass it on if its there